extensions/amp-subscriptions/0.1/platform-store.js

```diff
@@ -171,18 +235,61 @@ export class EntitlementStore {
   }
 
   /**
-   * Returns most qualified platform
-   * @param {!Array<!./entitlement.Entitlement>} entitlements
-   * @returns {!./entitlement.Entitlement}
+   * Returns most qualified platform.
+   * Qualification of a platform is based on an integer weight.
+   * Every platform starts with weight 0 and evaluated against the following parameters,
+   * - user is subscribed with platform (Gives weight 10)
+   * - supports the current viewer (Gives weight 9)
+   *
+   * In the end candidate with max weight is selected.
+   * However if candidate's weight is equal to local platform, then local platform is selected.
+   * @returns {!./subscription-platform.SubscriptionPlatform}
    */
-  selectApplicablePlatform_(entitlements) {
-    let chosenPlatform;
-    entitlements.forEach(platform => {
-      // TODO(@prateekbh): add metering logic here
-      if (platform.enablesThis()) {
-        chosenPlatform = platform;
+  selectApplicablePlatform_() {
+    const localPlatform = this.getLocalPlatform();
+
+    /** @type {!Object<string, number>} */
+    const platformWeights = {};
+
+    this.getAllResolvedPlatforms().forEach(platform => {
+      const entitlement =
+          this.getResolvedEntitlementFor(platform.getServiceId());
+
+      // Subscriber, gains weight 10
+      if (!!entitlement.subscriptionToken) {
+        platformWeights[platform.getServiceId()] = 10;
+      }
+
+      // If supports the current viewer, gains weight 9
+      if (platform.supportsCurrentViewer()) {
+        platformWeights[platform.getServiceId()] =
```

Comment:
Calculate via local var and assign to the map in the end when the score is ready.

Reply:
I dont really understand this one completely but, I guess i have flattened the algo a bit